phoenix-ios/phoenix-ios/views/configuration/privacy and security/app access/NumberPadView.swift

```diff
@@ -0,0 +1,230 @@
+	
+	enum NumberPadButtonWidth: Preference {}
+	let numberPadButtonWidthReader = GeometryPreferenceReader(
+		key: AppendValue<NumberPadButtonWidth>.self,
+		value: { [$0.size.width] }
+	)
+	@State var numberPadButtonWidth: CGFloat? = nil
+	
+	enum NumberPadButtonHeight: Preference {}
+	let numberPadButtonHeightReader = GeometryPreferenceReader(
+		key: AppendValue<NumberPadButtonHeight>.self,
+		value: { [$0.size.height] }
+	)
+	@State var numberPadButtonHeight: CGFloat? = nil
+	
+	// --------------------------------------------------
+	// MARK: View Builders
+	// --------------------------------------------------
+	
+	@ViewBuilder
+	var body: some View {
+		
+		VStack(alignment: HorizontalAlignment.trailing, spacing: 20) {
+			HStack(alignment: VerticalAlignment.center, spacing: 20) {
+				numberPadButton(._1)
+				numberPadButton(._2)
+				numberPadButton(._3)
+			}
+			HStack(alignment: VerticalAlignment.center, spacing: 20) {
+				numberPadButton(._4)
+				numberPadButton(._5)
+				numberPadButton(._6)
+			}
+			HStack(alignment: VerticalAlignment.center, spacing: 20) {
+				numberPadButton(._7)
+				numberPadButton(._8)
+				numberPadButton(._9)
+			}
+			HStack(alignment: VerticalAlignment.center, spacing: 20) {
+				if showHideButton {
+					numberPadButton(.hide)
+				}
+				numberPadButton(._0)
+				numberPadButton(.delete)
+			}
+		} // </VStack>
+		.assignMaxPreference(for: numberPadButtonWidthReader.key, to: $numberPadButtonWidth)
+		.assignMaxPreference(for: numberPadButtonHeightReader.key, to: $numberPadButtonHeight)
+	}
+	
+	@ViewBuilder
+	func numberPadButton(_ identifier: NumberPadButton) -> some View {
+		
+		if identifier == .hide {
+			
+			Button {
+				buttonPressed(identifier)
+			} label: {
+				Group {
+					if #available(iOS 17, *) {
+						Image(systemName: "arrowshape.down")
+							.resizable()
+							.aspectRatio(contentMode: .fit)
+					} else {
+						Image(systemName: "arrowtriangle.down")
+							.resizable()
+							.aspectRatio(contentMode: .fit)
+					}
+				}
+				.frame(width: 34, height: 34, alignment: .center)
+				.frame(width: buttonSize, height: buttonSize, alignment: .center)
+				.padding(.all, 10)
+			} // </Button>
+			.tint(.secondary)
+			
+		} else if identifier == .delete {
+			
+			Button {
+				buttonPressed(identifier)
+			} label: {
+				Image(systemName: "delete.backward")
+					.resizable()
+					.aspectRatio(contentMode: .fit)
+					.frame(width: 34, height: 34, alignment: .center)
+					.frame(width: buttonSize, height: buttonSize, alignment: .center)
+					.padding(.all, 10)
+			} // </Button>
+			.tint(.secondary)
+			.disabled(self.disabled)
+			
+		} else {
+			
+			Button {
+				buttonPressed(identifier)
+			} label: {
+				VStack(alignment: HorizontalAlignment.center, spacing: 2) {
+					
+					Text(verbatim: identifier.rawValue)
+						.font(.system(size: 34, weight: .medium))
+					
+					Group {
+						switch identifier {
+							case ._2 : Text(verbatim: "ABC")
```

Comment:
Why are there alphabetic characters in these buttons? IIUC the PIN is digits-only, and the UI does not let me long-press PIN buttons to use alphabetic chars instead of digits. Or am I missing something?

Reply:
Because the iOS system pin (numbers only) also displays the alphabetic characters:

<img height="450" src="https://github.com/ACINQ/phoenix/assets/304604/f7036d50-9f89-4b6b-b619-aea97eeb9717"/>

I assume Apple has always done it this way to help users create easy-to-remember PINS. E.g. "dios10" => 346710

I think Android might do it too, but I'm not sure about recent versions.

Reply:
However, this is probably only the case for languages that use a Latin alphabet... I'll update the code to only show the letters in that situation.

Reply:
Done in https://github.com/ACINQ/phoenix/pull/560/commits/2614dfed6fa1b44b7cc2c54ad0cc3860fc8faf58